src/main/java/executor/service/service/proxy/impl/ProxyValidatorImpl.java

```diff
@@ -0,0 +1,63 @@
+package executor.service.service.proxy.impl;
+
+import executor.service.config.properties.PropertiesConstants;
+import executor.service.model.proxy.ProxyConfigHolder;
+import executor.service.service.proxy.ProxyValidator;
+import org.apache.http.HttpHost;
+import org.apache.http.HttpStatus;
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+
+
+public class ProxyValidatorImpl implements ProxyValidator {
+    private final static String TARGET_URL = "http://www.google.com/";
+    private final static int CONNECTION_TIMEOUT = 10000;
+
+    public Boolean isValid(ProxyConfigHolder proxyConfigHolder) {
+        int responseCode = 0;
+
+        try {
+            CredentialsProvider credentialsProvider = getCredentialsProvider(proxyConfigHolder);
+            CloseableHttpClient httpClient = getHttpClient(proxyConfigHolder, credentialsProvider);
+
+            HttpGet httpGet = new HttpGet(TARGET_URL);
+            CloseableHttpResponse response = httpClient.execute(httpGet);
+            responseCode = response.getStatusLine().getStatusCode();
+
+            response.close();
+            httpClient.close();
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
Instead of **e.printStackTrace()**, it is recommended to log errors using SLF4J logging or another logging framework. This will make your code look more professional and help you track down bugs more easily.